src/pages/register-page.tsx

```diff
@@ -37,17 +50,153 @@ const styles = StyleSheet.create({
+    const userAuthIDAsyncStorageName = "@userAuthID";
+    const [inputValue, setInputValue] = useState('');
+    const [loginLoading, setLoginLoading] = useState(true);
     const [phoneNumber, setPhoneNumber] = useState('');
+    const [verificationID, setVerificationID] = useState<string | null>(null);
+    const [authenticationError, setAuthenticationError] = useState(false);
     const [showError, setError] = useState(false);
+    const recaptchaVerifier = React.useRef<FirebaseRecaptchaVerifierModal | null>(null);
+
+    useFocusEffect(
+        React.useCallback(() => {
+          const task = InteractionManager.runAfterInteractions(() => {
+            // Start login only when the screen is focused
+            if (props.router.navigation.isFocused()) {
+                // First we check if this this devices has already logged in before
+                AsyncStorage.getItem(userAuthIDAsyncStorageName).then(data => {
+                    if (data) {
+                        startLogin(data);
+                    }
+                    else {
+                        setLoginLoading(false);
+                    }
+                })
+            }
+          });
+      
+          return () => task.cancel();
+        }, [])
+      );
+
+    const checkPhoneNumber = () => {
+        setPhoneNumber(inputValue)
+        if (inputValue.length === numberLength) {
+            sendVerificationCode(inputValue);
+        }
+        else {
+            setError(true);
+        }
+    }
 
-    const checkPhoneNumber = () => {              //placeholder for actual function when using phone number to identify user
-        if (numberInput.length === 10) {
-            setPhoneNumber(numberInput)
-            navigation.navigate('Home');
+    const sendVerificationCode = async (number: string) => {
+        if (recaptchaVerifier.current) {
+            const phoneProvider = new firebase.auth.PhoneAuthProvider();
+            setInputValue('');
+            const verificationId = await phoneProvider.verifyPhoneNumber(
+                    inputValue,
+                    recaptchaVerifier.current
+            );
+            setVerificationID(verificationId);
         }
-        setError(numberInput.length !== 10)
+    }
+
+    const requestCredentials = async (verificationCode: string, verificationID: string) => {
+        setLoginLoading(true);
+        try {
+            // Try to verify the credentials
+            const credential = firebase.auth.PhoneAuthProvider.credential(
+                verificationID,
+                verificationCode
+            );
+
+            // Get a auth user
+            const userAuth = await firebase.auth().signInWithCredential(credential);
+            const userID = userAuth.user?.uid;
+
+            // Saving the user and continue to login
+            if (userID) {
+                await AsyncStorage.setItem(userAuthIDAsyncStorageName, userID);
+                await startLogin(userAuthIDAsyncStorageName);
```

Comment:
```suggestion
                await startLogin(userID);
```
I think this was the problem!

Reply:
Ahh, yea, that looks wrong